resteasy-reactive-jaxb does not work nicely together with multipart/form-data FileUpload

### Describe the bug

I have 3 resteasy-reactive resources in my project:
- one which returns application/xml
- one which consumes multipart/form-data. This uses a org.jboss.resteasy.reactive.multipart.FileUpload.
- one which returns text/plain

The text/plain one works correctly.
The multipart/form-data one also works correctly.

The application/xml one however complains about the FileUpload which is used in the other resource. See exception below.
FileUpload  is an Interface, and can't be marshalled / unmarshaled by default.

The FileUpload class is bound because each  parameter type of a multipart/form-data resource is getting bound to one jaxb context.
https://github.com/quarkusio/quarkus/blob/b9d7536effc07f08f119eea292673da6e0826057/extensions/resteasy-reactive/quarkus-resteasy-reactive-jaxb/deployment/src/main/java/io/quarkus/resteasy/reactive/jaxb/deployment/ResteasyReactiveJaxbProcessor.java#L89-L98

One solution would be to create distinct JaxbContext instances for each resource.

Using java.io.File instead works, however in that case I can't access the  mimetype easily.

### Expected behavior

No exception when using multipart/form-data and application/xml in the same project.

### Actual behavior

```
2022-10-12 08:28:22,045 ERROR [org.jbo.res.rea.com.cor.AbstractResteasyReactiveContext] (executor-thread-0) Request failed: java.lang.RuntimeException: com.sun.xml.bind.v2.runtime.IllegalAnnotationsException: 1 counts of IllegalAnnotationExceptions
org.jboss.resteasy.reactive.multipart.FileUpload ist eine Schnittstelle, und JAXB kann keine Schnittstellen verarbeiten.
        this problem is related to the following location:
                at org.jboss.resteasy.reactive.multipart.FileUpload
                at public org.jboss.resteasy.reactive.multipart.FileUpload org.acme.FileUploadData.fileUpload
                at org.acme.FileUploadData

        at io.quarkus.jaxb.runtime.JaxbContextProducer.jaxbContext(JaxbContextProducer.java:44)
        at io.quarkus.jaxb.runtime.JaxbContextProducer_Subclass.jaxbContext$$superforward1(Unknown Source)
        at io.quarkus.jaxb.runtime.JaxbContextProducer_Subclass$$function$$3.apply(Unknown Source)
        at io.quarkus.arc.impl.AroundInvokeInvocationContext.proceed(AroundInvokeInvocationContext.java:54)
        at io.quarkus.arc.runtime.devconsole.InvocationInterceptor.proceed(InvocationInterceptor.java:62)
        at io.quarkus.arc.runtime.devconsole.InvocationInterceptor.monitor(InvocationInterceptor.java:49)
        at io.quarkus.arc.runtime.devconsole.InvocationInterceptor_Bean.intercept(Unknown Source)
        at io.quarkus.arc.impl.InterceptorInvocation.invoke(InterceptorInvocation.java:42)
        at io.quarkus.arc.impl.AroundInvokeInvocationContext.perform(AroundInvokeInvocationContext.java:41)
        at io.quarkus.arc.impl.InvocationContexts.performAroundInvoke(InvocationContexts.java:33)
        at io.quarkus.jaxb.runtime.JaxbContextProducer_Subclass.jaxbContext(Unknown Source)
        at io.quarkus.jaxb.runtime.JaxbContextProducer_ProducerMethod_jaxbContext_bd8f21cf3f403916a32bb5db1dc1007aab351487_Bean.create(Unknown Source)
        at io.quarkus.jaxb.runtime.JaxbContextProducer_ProducerMethod_jaxbContext_bd8f21cf3f403916a32bb5db1dc1007aab351487_Bean.create(Unknown Source)
        at io.quarkus.arc.impl.AbstractSharedContext.createInstanceHandle(AbstractSharedContext.java:113)
        at io.quarkus.arc.impl.AbstractSharedContext$1.get(AbstractSharedContext.java:37)
        at io.quarkus.arc.impl.AbstractSharedContext$1.get(AbstractSharedContext.java:34)
        at io.quarkus.arc.impl.LazyValue.get(LazyValue.java:26)
        at io.quarkus.arc.impl.ComputingCache.computeIfAbsent(ComputingCache.java:69)
        at io.quarkus.arc.impl.AbstractSharedContext.get(AbstractSharedContext.java:34)
        at io.quarkus.jaxb.runtime.JaxbContextProducer_ProducerMethod_jaxbContext_bd8f21cf3f403916a32bb5db1dc1007aab351487_Bean.get(Unknown Source)
        at io.quarkus.jaxb.runtime.JaxbContextProducer_ProducerMethod_jaxbContext_bd8f21cf3f403916a32bb5db1dc1007aab351487_Bean.get(Unknown Source)
        at io.quarkus.jaxb.runtime.JaxbContextProducer_ProducerMethod_marshaller_4f8d16bb162b44aa8bbc0b7b364931bb173b516e_Bean.create(Unknown Source)
        at io.quarkus.jaxb.runtime.JaxbContextProducer_ProducerMethod_marshaller_4f8d16bb162b44aa8bbc0b7b364931bb173b516e_Bean.create(Unknown Source)
        at io.quarkus.arc.impl.RequestContext.getIfActive(RequestContext.java:76)
        at io.quarkus.arc.impl.ClientProxies.getDelegate(ClientProxies.java:30)
        at javax.xml.bind.JaxbContextProducer_ProducerMethod_marshaller_4f8d16bb162b44aa8bbc0b7b364931bb173b516e_ClientProxy.arc$delegate(Unknown Source)
        at javax.xml.bind.JaxbContextProducer_ProducerMethod_marshaller_4f8d16bb162b44aa8bbc0b7b364931bb173b516e_ClientProxy.marshal(Unknown Source)
        at io.quarkus.resteasy.reactive.jaxb.runtime.serialisers.ServerJaxbMessageBodyWriter.marshal(ServerJaxbMessageBodyWriter.java:57)
        at io.quarkus.resteasy.reactive.jaxb.runtime.serialisers.ServerJaxbMessageBodyWriter.writeResponse(ServerJaxbMessageBodyWriter.java:43)
        at org.jboss.resteasy.reactive.server.core.ServerSerialisers.invokeWriter(ServerSerialisers.java:216)
        at org.jboss.resteasy.reactive.server.core.ServerSerialisers.invokeWriter(ServerSerialisers.java:187)
        at org.jboss.resteasy.reactive.server.core.serialization.FixedEntityWriter.write(FixedEntityWriter.java:28)
        at org.jboss.resteasy.reactive.server.handlers.ResponseWriterHandler.handle(ResponseWriterHandler.java:34)
        at io.quarkus.resteasy.reactive.server.runtime.QuarkusResteasyReactiveRequestContext.invokeHandler(QuarkusResteasyReactiveRequestContext.java:121)
        at org.jboss.resteasy.reactive.common.core.AbstractResteasyReactiveContext.run(AbstractResteasyReactiveContext.java:142)
        at io.quarkus.vertx.core.runtime.VertxCoreRecorder$14.runWith(VertxCoreRecorder.java:564)
        at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2449)
        at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1478)
        at org.jboss.threads.DelegatingRunnable.run(DelegatingRunnable.java:29)
        at org.jboss.threads.ThreadLocalResettingRunnable.run(ThreadLocalResettingRunnable.java:29)
        at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
        at java.base/java.lang.Thread.run(Thread.java:833)
Caused by: com.sun.xml.bind.v2.runtime.IllegalAnnotationsException: 1 counts of IllegalAnnotationExceptions
org.jboss.resteasy.reactive.multipart.FileUpload ist eine Schnittstelle, und JAXB kann keine Schnittstellen verarbeiten.
        this problem is related to the following location:
                at org.jboss.resteasy.reactive.multipart.FileUpload
                at public org.jboss.resteasy.reactive.multipart.FileUpload org.acme.FileUploadData.fileUpload
                at org.acme.FileUploadData

        at com.sun.xml.bind.v2.runtime.IllegalAnnotationsException$Builder.check(IllegalAnnotationsException.java:76)
        at com.sun.xml.bind.v2.runtime.JAXBContextImpl.getTypeInfoSet(JAXBContextImpl.java:441)
        at com.sun.xml.bind.v2.runtime.JAXBContextImpl.<init>(JAXBContextImpl.java:273)
        at com.sun.xml.bind.v2.runtime.JAXBContextImpl.<init>(JAXBContextImpl.java:109)
        at com.sun.xml.bind.v2.runtime.JAXBContextImpl$JAXBContextBuilder.build(JAXBContextImpl.java:1126)
        at com.sun.xml.bind.v2.ContextFactory.createContext(ContextFactory.java:135)
        at com.sun.xml.bind.v2.JAXBContextFactory.createContext(JAXBContextFactory.java:35)
        at javax.xml.bind.ContextFinder.find(ContextFinder.java:393)
        at javax.xml.bind.JAXBContext.newInstance(JAXBContext.java:691)
        at io.quarkus.jaxb.runtime.JaxbContextProducer.jaxbContext(JaxbContextProducer.java:42)
        ... 41 more
```

### How to Reproduce?

1. Download the reproducer: 
[multipart-jaxb-fileinput.zip](https://github.com/quarkusio/quarkus/files/9762200/multipart-jaxb-fileinput.zip)
2. Either execute the Test GreetingResourceTest
3. Or mvn quarkus:dev and visit http://localhost:8080/hello/xml

### Output of `uname -a` or `ver`

Microsoft Windows [Version 10.0.19044.2006]

### Output of `java -version`

17.0.4

### GraalVM version (if different from Java)

_No response_

### Quarkus version or git rev

2.13.1.Final

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

Apache Maven 3.8.4 (9b656c72d54e5bacbed989b64718c159fe39b537) Maven home: C:\eclipse\tools\java\maven Java version: 17.0.4, vendor: Eclipse Adoptium, runtime: C:\eclipse\tools\java\17 Default locale: de_DE, platform encoding: Cp1252 OS name: "windows 10", version: "10.0", arch: "amd64", family: "windows"

### Additional information

_No response_

--- a/extensions/resteasy-reactive/quarkus-resteasy-reactive-jaxb/deployment/src/main/java/io/quarkus/resteasy/reactive/jaxb/deployment/ResteasyReactiveJaxbProcessor.java
+++ b/extensions/resteasy-reactive/quarkus-resteasy-reactive-jaxb/deployment/src/main/java/io/quarkus/resteasy/reactive/jaxb/deployment/ResteasyReactiveJaxbProcessor.java
@@ -39,6 +39,8 @@ import io.quarkus.resteasy.reactive.spi.MessageBodyWriterBuildItem;
 
 public class ResteasyReactiveJaxbProcessor {
 
+    private static final List<String> XML_TYPES = List.of(MediaType.APPLICATION_XML, MediaType.TEXT_XML);
+
     @BuildStep
     void feature(BuildProducer<FeatureBuildItem> feature) {
         feature.produce(new FeatureBuildItem(Feature.RESTEASY_REACTIVE_JAXB));
@@ -56,10 +58,10 @@ public class ResteasyReactiveJaxbProcessor {
 
         additionalReaders
                 .produce(new MessageBodyReaderBuildItem(ServerJaxbMessageBodyReader.class.getName(), Object.class.getName(),
-                        List.of(MediaType.APPLICATION_XML, MediaType.TEXT_XML), RuntimeType.SERVER, true, Priorities.USER));
+                        XML_TYPES, RuntimeType.SERVER, true, Priorities.USER));
         additionalWriters
                 .produce(new MessageBodyWriterBuildItem(ServerJaxbMessageBodyWriter.class.getName(), Object.class.getName(),
-                        List.of(MediaType.APPLICATION_XML, MediaType.TEXT_XML), RuntimeType.SERVER, true, Priorities.USER));
+                        XML_TYPES, RuntimeType.SERVER, true, Priorities.USER));
     }
 
     @BuildStep
@@ -87,11 +89,17 @@ public class ResteasyReactiveJaxbProcessor {
             }
 
             // If consumes "application/xml" or "multipart/form-data", we register all the classes of the parameters
-            if (consumesXml(resourceInfo) || consumesMultipart(resourceInfo)) {
+            boolean consumesXml = consumesXml(resourceInfo);
+            boolean consumesMultipart = consumesMultipart(resourceInfo);
+            if (consumesXml || consumesMultipart) {
                 for (Type parameter : methodInfo.parameterTypes()) {
                     ClassInfo effectiveParameter = getEffectiveClassInfo(parameter, indexView);
                     if (effectiveParameter != null) {
-                        classesInfo.add(effectiveParameter);
+                        if (consumesXml) {
+                            classesInfo.add(effectiveParameter);
+                        } else if (consumesMultipart) {
+                            classesInfo.addAll(getEffectivePartsUsingXml(effectiveParameter, indexView));
+                        }
                     }
                 }
             }
@@ -112,16 +120,24 @@ public class ResteasyReactiveJaxbProcessor {
     private List<ClassInfo> getEffectivePartsUsingXml(ClassInfo returnType, IndexView indexView) {
         List<ClassInfo> classInfos = new ArrayList<>();
         for (FieldInfo field : returnType.fields()) {
-            AnnotationInstance partTypeInstance = field.annotation(ResteasyReactiveDotNames.PART_TYPE_NAME);
-            if (partTypeInstance != null) {
-                AnnotationValue partTypeValue = partTypeInstance.value();
-                if (partTypeValue != null && MediaType.APPLICATION_XML.equals(partTypeValue.asString())) {
-                    classInfos.add(getEffectiveClassInfo(field.type(), indexView));
-                }
+            if (isPartTypeXml(field)) {
+                classInfos.add(getEffectiveClassInfo(field.type(), indexView));
             }
         }
 
         return classInfos;
+    }
+
+    private boolean isPartTypeXml(FieldInfo field) {
+        AnnotationInstance partType = field.annotation(ResteasyReactiveDotNames.PART_TYPE_NAME);
+        if (partType != null) {
+            AnnotationValue partTypeValue = partType.value();
+            if (containsMediaType(new String[] { partTypeValue.asString() }, XML_TYPES)) {
+                return true;
+            }
+        }
+
+        return false;
     }
 
     private ClassInfo getEffectiveClassInfo(Type type, IndexView indexView) {
@@ -158,26 +174,28 @@ public class ResteasyReactiveJaxbProcessor {
     }
 
     private boolean consumesXml(ResourceMethod resourceInfo) {
-        return containsMediaType(resourceInfo.getConsumes(), MediaType.APPLICATION_XML);
+        return containsMediaType(resourceInfo.getConsumes(), XML_TYPES);
     }
 
     private boolean consumesMultipart(ResourceMethod resourceInfo) {
-        return containsMediaType(resourceInfo.getConsumes(), MediaType.MULTIPART_FORM_DATA);
+        return containsMediaType(resourceInfo.getConsumes(), List.of(MediaType.MULTIPART_FORM_DATA));
     }
 
     private boolean producesXml(ResourceMethod resourceInfo) {
-        return containsMediaType(resourceInfo.getProduces(), MediaType.APPLICATION_XML);
+        return containsMediaType(resourceInfo.getProduces(), XML_TYPES);
     }
 
     private boolean producesMultipart(ResourceMethod resourceInfo) {
-        return containsMediaType(resourceInfo.getProduces(), MediaType.MULTIPART_FORM_DATA);
+        return containsMediaType(resourceInfo.getProduces(), List.of(MediaType.MULTIPART_FORM_DATA));
     }
 
-    private boolean containsMediaType(String[] types, String mediaType) {
+    private boolean containsMediaType(String[] types, List<String> mediaTypes) {
         if (types != null) {
             for (String type : types) {
-                if (type.toLowerCase(Locale.ROOT).contains(mediaType)) {
-                    return true;
+                for (String mediaType : mediaTypes) {
+                    if (type.toLowerCase(Locale.ROOT).contains(mediaType)) {
+                        return true;
+                    }
                 }
             }
         }

--- a/extensions/resteasy-reactive/quarkus-resteasy-reactive-jaxb/deployment/src/test/java/io/quarkus/resteasy/reactive/jaxb/deployment/test/MultipartTest.java
+++ b/extensions/resteasy-reactive/quarkus-resteasy-reactive-jaxb/deployment/src/test/java/io/quarkus/resteasy/reactive/jaxb/deployment/test/MultipartTest.java
@@ -2,7 +2,12 @@ package io.quarkus.resteasy.reactive.jaxb.deployment.test;
 
 import static org.assertj.core.api.Assertions.assertThat;
 
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+
 import javax.ws.rs.Consumes;
+import javax.ws.rs.FormParam;
 import javax.ws.rs.GET;
 import javax.ws.rs.POST;
 import javax.ws.rs.Path;
@@ -12,6 +17,7 @@ import javax.ws.rs.core.MediaType;
 import org.jboss.resteasy.reactive.MultipartForm;
 import org.jboss.resteasy.reactive.PartType;
 import org.jboss.resteasy.reactive.RestForm;
+import org.jboss.resteasy.reactive.multipart.FileUpload;
 import org.jboss.shrinkwrap.api.ShrinkWrap;
 import org.jboss.shrinkwrap.api.spec.JavaArchive;
 import org.junit.jupiter.api.Test;
@@ -36,6 +42,7 @@ public class MultipartTest {
             + "<school>"
             + "<name>Divino Pastor</name>"
             + "</school>";
+    private final File HTML_FILE = new File("./src/test/resources/test.html");
 
     @RegisterExtension
     static QuarkusUnitTest test = new QuarkusUnitTest()
@@ -51,7 +58,7 @@ public class MultipartTest {
                 .extract().asString();
 
         assertContains(response, "name", MediaType.TEXT_PLAIN, EXPECTED_RESPONSE_NAME);
-        assertContains(response, "person", MediaType.APPLICATION_XML, EXPECTED_RESPONSE_PERSON);
+        assertContains(response, "person", MediaType.TEXT_XML, EXPECTED_RESPONSE_PERSON);
     }
 
     @Test
@@ -66,6 +73,19 @@ public class MultipartTest {
                 .extract().asString();
 
         assertThat(response).isEqualTo("John-Divino Pastor");
+    }
+
+    @Test
+    public void testInputFile() throws IOException {
+        String response = RestAssured
+                .given()
+                .multiPart("file", HTML_FILE, "text/html")
+                .post("/multipart/input/file")
+                .then()
+                .statusCode(200)
+                .extract().asString();
+
+        assertThat(response).isEqualTo(String.valueOf(Files.readAllBytes(HTML_FILE.toPath()).length));
     }
 
     private void assertContains(String response, String name, String contentType, Object value) {
@@ -97,6 +117,13 @@ public class MultipartTest {
             return input.name + "-" + input.school.name;
         }
 
+        @POST
+        @Path("/input/file")
+        @Consumes(MediaType.MULTIPART_FORM_DATA)
+        public int inputFile(@MultipartForm FileUploadData data) throws IOException {
+            return Files.readAllBytes(data.fileUpload.filePath()).length;
+        }
+
     }
 
     private static class MultipartOutputResponse {
@@ -105,8 +132,13 @@ public class MultipartTest {
         String name;
 
         @RestForm
-        @PartType(MediaType.APPLICATION_XML)
+        @PartType(MediaType.TEXT_XML)
         Person person;
+    }
+
+    public static class FileUploadData {
+        @FormParam("file")
+        FileUpload fileUpload;
     }
 
     public static class MultipartInput {

--- a/extensions/resteasy-reactive/quarkus-resteasy-reactive-jaxb/deployment/src/test/resources/test.html
+++ b/extensions/resteasy-reactive/quarkus-resteasy-reactive-jaxb/deployment/src/test/resources/test.html
@@ -1,0 +1,9 @@
+<!DOCTYPE html>
+<html>
+<head>
+    <!-- head definitions go here -->
+</head>
+<body>
+<!-- the content goes here -->
+</body>
+</html>

